skimage/restoration/_nl_means_denoising.pyx

```diff
@@ -0,0 +1,596 @@
+import numpy as np
+from skimage import util
+cimport numpy as np
+cimport cython
+from libc.math cimport exp
+
+ctypedef np.float32_t DTYPE_t
+
+cdef eps = 1.e-8
+
+
+@cython.boundscheck(False)
+cdef inline float patch_distance_2d(DTYPE_t [:, :] p1,
+                                    DTYPE_t [:, :] p2,
+                                    DTYPE_t [:, ::] w, int s):
+    cdef int i, j
+    cdef int center = s / 2
+    # Check if central pixel is too different in the 2 patches
+    cdef float tmp_diff = p1[center, center] - p2[center, center]
+    cdef float init = w[center, center] * tmp_diff * tmp_diff
+    if init > 1:
+        return eps
+    cdef float distance = 0
+    for i in range(s):
+        # exp of large negative numbers will be 0, so we'd better stop
+        if distance > 5:
```

Comment:
Thanks for noticing this inconsistency. In fact, setting ``eps`` to a non-zero value was just a trick to avoid divisions by zero, but I think there is no risk of division by zero. We could then just return 0. The idea of such a cut-off for distance is that is the two patches are sufficiently different from each other, then there is no good reason to average the central pixels together. And this condition saves in fact a lot of computing time.